### PR TITLE
Replacing constructors with static factory methods

### PR DESCRIPTION
**Pull request by zoewangg**
## Description
Replacing constructors with static factory methods. Note, this only contains refactoring for classes under `software.amazon.awssdk.core.auth` For other classes, we decided to leave them as it is since some of the classes are just for internal use and some will be deprecated.
Created the wiki page in my forked repo https://github.com/zoewangg/aws-sdk-java-v2/wiki/Favor-Static-Factory-Methods-Over-Constructors. If it looks good, I'll move the wiki page to this repo.

## Testing
`mvn clean install`

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [x] I have read the **README** document
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: core/src/main/java/software/amazon/awssdk/core/auth/AbstractAwsSigner.java [hunk 1]
```diff
@@ -316,12 +316,12 @@ protected AwsCredentials sanitizeCredentials(AwsCredentials credentials) {
 
         if (credentials instanceof AwsSessionCredentials) {
             AwsSessionCredentials sessionCredentials = (AwsSessionCredentials) credentials;
-            return new AwsSessionCredentials(accessKeyId,
+            return AwsSessionCredentials.create(accessKeyId,
                                              secretKey,
```
Comment (spfink) [inline core/src/main/java/software/amazon/awssdk/core/auth/AbstractAwsSigner.java]: Alignment of parameters is off here

File: core/src/main/java/software/amazon/awssdk/core/auth/presign/PresignerParams.java [hunk 2]
```diff
@@ -35,7 +35,7 @@
     private final SignerProvider signerProvider;
     private final SdkClock clock;
 
-    public PresignerParams(URI endpoint,
+    private PresignerParams(URI endpoint,
                            AwsCredentialsProvider credentialsProvider,
```
Comment (spfink) [inline core/src/main/java/software/amazon/awssdk/core/auth/presign/PresignerParams.java]: Alignment of parameters

File: core/src/test/java/software/amazon/awssdk/core/AmazonWebServiceRequestTest.java [hunk 3]
```diff
@@ -91,7 +91,7 @@ public void copyBaseTo() {
             public void progressChanged(ProgressEvent progressEvent) {
             }
         };
-        final AwsCredentials credentials = new AwsCredentials("accesskey",
+        final AwsCredentials credentials = AwsCredentials.create("accesskey",
                                                               "accessid");
```
Comment (spfink) [inline core/src/test/java/software/amazon/awssdk/core/AmazonWebServiceRequestTest.java]: Alignment

File: core/src/main/java/software/amazon/awssdk/core/auth/AbstractAwsSigner.java (hunk 1 above)
Comment (zoewangg) [reply core/src/main/java/software/amazon/awssdk/core/auth/AbstractAwsSigner.java]: my bad. fixed all of them

**Pull request closed without merging**